src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -168,6 +168,19 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
             Visit(node.Body);
         }
 
+        public override void VisitRazorMetaCode(RazorMetaCodeSyntax node)
+        {
+            switch (node.Kind)
+            {
+                // These are Braces for things like "@code {}" and "@functions {}".
+                // Hypothetically the braces are the only metacode which should ever reach this,
+                // but I added the switch statement just in case.
```

Comment:
I suggest removing the switch statement here. We want to know if the node type and kind at mismatched (which should never happen). Plus if we have this assumption in so many places already.

Reply:
Ya we should be exploding so we can properly handle bad situations instead of hiding it.